### PR TITLE
[codemod] Avoid filter failures on object prototype properties

### LINKED ISSUE #8625: [codemod] Undefined does not match field "name": string of type Identifier when running @mui/x-codemod v6.0.0/preset-safe
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example:

Steps:
1. Install the latest mui-v6 date and datagrid packages
2. Run the codemod using `npx @mui/x-codemod v6.0.0/preset-safe <path>`

### Current behavior 😯

The codemod fails consistently (and all it's variations, aka trying to run a single transformation instead of all of them), with the following stacktrace

```
ERR /Users/jeetmehta/Developer/swift/components/admin/calendar/CalendarView.tsx Transformation error (undefined does not match field "name": string of type Identifier)
Error: undefined does not match field "name": string of type Identifier
    at addParam (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/ast-types/lib/types.js:446:27)
    at /Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/ast-types/lib/types.js:469:25
    at Array.forEach (<anonymous>)
    at Function.identifier (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/ast-types/lib/types.js:467:35)
    at NodePath.<anonymous> (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/v6.0.0/pickers/view-components-rename/index.js:81:93)
    at NodePath.<anonymous> (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/node_modules/jscodeshift/src/collections/Node.js:142:47)
    at /Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/node_modules/jscodeshift/src/Collection.js:75:36
    at Array.forEach (<anonymous>)
    at Collection.forEach (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/node_modules/jscodeshift/src/Collection.js:74:18)
    at Collection.replaceWith (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/node_modules/jscodeshift/src/collections/Node.js:140:17)
 ERR /Users/jeetmehta/Developer/swift/components/book/memberships/MembershipsCard.tsx Transformation error (undefined does not match field "name": string of type Identifier)
Error: undefined does not match field "name": string of type Identifier
    at addParam (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/ast-types/lib/types.js:446:27)
    at /Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/ast-types/lib/types.js:469:25
    at Array.forEach (<anonymous>)
    at Function.identifier (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/ast-types/lib/types.js:467:35)
    at NodePath.<anonymous> (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/v6.0.0/pickers/view-components-rename/index.js:81:93)
    at NodePath.<anonymous> (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/node_modules/jscodeshift/src/collections/Node.js:142:47)
    at /Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/node_modules/jscodeshift/src/Collection.js:75:36
    at Array.forEach (<anonymous>)
    at Collection.forEach (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/node_modules/jscodeshift/src/Collection.js:74:18)
    at Collection.replaceWith (/Users/jeetmehta/.npm/_npx/a9b616de167a2c5b/node_modules/@mui/x-codemod/node_modules/jscodeshift/src/collections/Node.js:140:17)
```

### Expected behavior 🤔

Ideally, the codemod should run without errors and apply the necessary modifications to the codebase

### Context 🔦

I'm trying to upgrade our MUI date pickers and datagrid components, to be able to use the new TimePicker component. We use MUI extensively throughout our codebase, so being forced to manually migrate all our components is going to be a real pain. We'd really like the codemod to help with this.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
System:
    OS: macOS 13.3
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.6.4 - ~/.nvm/versions/node/v18.12.1/bin/npm
  Browsers:
    Chrome: 112.0.5615.49
    Edge: Not Found
    Firefox: 106.0.1
    Safari: 16.4
  npmPackages:
    @emotion/react:  11.10.5
    @emotion/styled: ^11.10.5 => 11.10.5
    @mui/base:  5.0.0-alpha.118
    @mui/core-downloads-tracker:  5.11.9
    @mui/icons-material: ^5.11.9 => 5.11.9
    @mui/lab: ^5.0.0-alpha.120 => 5.0.0-alpha.120
    @mui/material: ^5.11.9 => 5.11.9
    @mui/private-theming:  5.11.9
    @mui/styled-engine:  5.11.9
    @mui/system:  5.11.9
    @mui/types:  7.2.3
    @mui/utils:  5.11.9
    @mui/x-data-grid: next => 6.0.0-beta.5
    @mui/x-date-pickers: next => 6.0.0-beta.5
    @types/react: ^18.0.28 => 18.0.28
    react: ^18.2.0 => 18.2.0
    react-dom: ^18.2.0 => 18.2.0
    styled-components: ^5.3.6 => 5.3.6
    typescript: ^4.9.5 => 4.9.5

Used Chrome for development, but this error occurs in the console when running the codemod so it doesn't really matter.
```
</details>

### Order ID or Support key 💳 (optional)

_No response_

### PR DESCRIPTION
**Pull request by LukasTy**
Fixes #8625 

Filters were not excluding edge cases like `VARIABLES['toString']` which does return an object prototype function.

Added test to cover the mentioned case as well as include data grid usage in `preset-safe` test to catch the `data-grid` transformer issues.

### CODE REVIEW

File: packages/x-codemod/src/util/renameIdentifiers.ts [hunk 1]
```diff
@@ -129,7 +129,7 @@ export default function renameIdentifiers({
 }: RenameIdentifiersArgs) {
   root
     .find(j.Identifier)
-    .filter((path) => !!identifiers[path.node.name])
+    .filter((path) => Object.hasOwn(identifiers, path.node.name))
```
Comment (LukasTy) [inline packages/x-codemod/src/util/renameIdentifiers.ts:132]: What do you think about using [this new API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn#browser_compatibility) which causes to bump the node engine version requirement? 🤔 
- Leave it for the next major? (probably will not be updated to use this new API in that case 🙈 )
- Go with the node engine bump as the codemod is only used by developers anyways, and possibly very small amount of them are stuck only on `< v16.9.0` 🤔

Comment (LukasTy) [reply packages/x-codemod/src/util/renameIdentifiers.ts:132]: Ok ok... Nevermind. 🙈 
Our CI is strictly node 14 and until that is updated, it won't build the package. 🤦

**Review by flaviendelangle — APPROVED** [review]
Very nice catch !
